### PR TITLE
ddl: treat the gc too early error retryable while adding index

### LINKED ISSUE #40074: GC life time is shorter than transaction duration while running DDL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Run add index job in a large table, and set the batch_size to 10240.

It is not easy to reproduce because the DDL txn is usually fast.

### 2. What did you expect to see? (Required)

No error.

### 3. What did you see instead (Required)

Ingest add index job fallback to txn-merge mode. 
```
[INFO] [index.go:762] ["[ddl] fallback to txn-merge backfill process"] [error="[tikv:9006]GC life time is shorter than transaction duration, transaction starts at 2022-12-20 10:58:20.94 +0000 UTC, GC safe point is 2022-12-20 10:58:22.14 +0000 UTC"]
```

### 4. What is your TiDB version? (Required)

ed33fdea6

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #40074

Problem Summary:

The txns created by DDL are not aware by the GC safepoint updater. Moreover, the txns created by adding index are usually large and it takes for a while to run.

### What is changed and how it works?

To make the risk under control, this PR treats these error as retryable so that it won't fallback to the txn-merge mode.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ import (
 	"github.com/pingcap/tidb/parser/model"
 	"github.com/pingcap/tidb/parser/mysql"
 	"github.com/pingcap/tidb/sessionctx"
+	tikverror "github.com/pingcap/tidb/store/driver/error"
```
Comment (hawkingrei) [inline ddl/index.go:42]: Remember to update the bazel config.

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request closed without merging**